File: src/string/sortString/test_sort_string.py
from sort_string import *
import pytest


@pytest.mark.parametrize('input, expected', [
    ('', ''),
    ('a', 'a'),
    ('ab', 'ab'),
    ('a!a', '!aa'),
    ('a b', ' ab'),
    ('ba b', ' abb'),
    ('ёе', 'её')
])
def test_sort_string(input, expected):
    assert sort_string(input) == expected

File: src/string/sortString/TestSortString.java
package string.sortString;

import static org.junit.Assert.assertEquals;
import static string.sortString.SortString.sortString;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

public class TestSortString {

    @ParameterizedTest
    // @MethodSource("provideParams")
    @CsvSource(value = {
            "' ', ' '",
            "'a', 'a'",
            "'ab', 'ab'",
            "'a!a', '!aa'",
            "'a b', ' ab'",
            "'ba b', ' abb'",
            "'ёе', 'её'"
    })
    void testSortString(String input, String expected) {
        assertEquals(expected, sortString(input));
    }

}